data/json/mapgen/Metalworker.json

```diff
@@ -0,0 +1,114 @@
+[
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [ "craft_shop" ],
+    "weight": 200,
+    "object": {
+      "fill_ter": "t_strconc_floor",
+      "rows": [
+        "__________8888888888888_",
+        "__________8___________8_",
+        "__________8_|--||--||_8_",
+        "_8888^^^^^8_-P.k...P|_8_",
+        "_8__________+..kh..N|_8_",
+        "_8__________-..kk.||||8_",
+        "_8________s_|.....|.t|8_",
+        "_8__|=====%||N....+.S|8_",
+        "_8__|.....sLLR...|||||8_",
+        "_8__|..............q|_8_",
+        "_8__|..........Q...Y|_8_",
+        "_8__|........Q.Q...Y|_8_",
+        "_8__|q.......Q.Q.#.m|_8_",
+        "_8__|q.......Q.Q....|_8_",
+        "_8__|q.............P|_8_",
+        "_8_U|........?||+-|||_8_",
+        "_8_U|R....sRm?|...HHP_8_",
+        "_8__|=====%|--|......_8_",
+        "_8________s___________8_",
+        "_8_v1__________v3v____8_",
+        "_8_vv________vvvvv___28_",
+        "_8_____________1_____28_",
+        "_8888888888888888888888_",
+        "________________________"
+      ],
+      "terrain": {
+        "s": "t_gates_control_concrete",
+        "=": "t_door_metal_locked",
+        "8": "t_chainfence",
+        "-": "t_wall_glass",
+        "|": "t_scrap_wall",
+        "%": "t_wall",
+        "v": "t_junk_wall",
+        "^": "t_chaingate_l",
+        "+": "t_door_glass_c",
+        "_": "t_rock_floor",
+        "U": "t_rock_floor",
+        "2": "t_rock_floor",
+        ".": "t_strconc_floor",
+        "k": "t_strconc_floor",
+        "h": "t_strconc_floor",
+        "P": "t_strconc_floor",
+        "R": "t_strconc_floor",
+        "L": "t_strconc_floor",
+        "Q": "t_strconc_floor",
+        "Y": "t_strconc_floor",
+        "q": "t_strconc_floor",
+        "t": "t_strconc_floor",
+        "S": "t_strconc_floor",
+        "?": "t_strconc_floor",
+        "m": "t_strconc_floor",
+        "N": "t_strconc_floor",
+        "H": "t_strconc_floor",
```

Comment:
You could cut a lot of lines out here, since these are all your fill_ter so that's what's going to be under them anyway